### ISSUE TITLE
Entities relations with invalid associations shown

### ISSUE THREAD

**Issue by paulo-graca** [DSpace/dspace-angular#1387]
**Describe the bug**
My model has two relation types, with same relation name, but involving different entities:
```
	<type>
		<leftType>Publication</leftType>
		<rightType>Person</rightType>
		<leftwardType>isAuthorOfPublication</leftwardType>
		<rightwardType>isPublicationOfAuthor</rightwardType>
		<leftCardinality>
			<min>0</min>
		</leftCardinality>
		<rightCardinality>
			<min>0</min>
		</rightCardinality>
	</type>
	<type>
		<leftType>Publication</leftType>
		<rightType>OrgUnit</rightType>
		<leftwardType>isAuthorOfPublication</leftwardType>
		<rightwardType>isPublicationOfAuthor</rightwardType>
		<leftCardinality>
			<min>0</min>
		</leftCardinality>
		<rightCardinality>
			<min>0</min>
		</rightCardinality>
	</type>
```
But when I try to associate a Person as an author, that Person, will be shown simultaneously as `Authors (persons)` relation and `Authors (organizational units)`

**To Reproduce**
1. You can use `/dspace/config/entities/openaire4-relationships.xml` as your model
2. And set `dspace.​entity.​type = Publication` to any existing Item
3. At edit item `Relationships` separator, use lookup to associate an author
4. After you added the relations to that entity, strange associations will be shown
![image](https://user-images.githubusercontent.com/877488/139826759-85d70200-3448-4a28-b728-f99a6940d890.png)

And if you refresh the page, you will get 2 associations:
 ![image](https://user-images.githubusercontent.com/877488/139819099-77ad95da-6a7f-4633-bec3-c4e44461ad3d.png)

It seems an issue with Relationships separator, because item page is shown as expected.

**Expected behavior**
I was expecting to have only one association, or to be shown only one `Authors (persons)` relation.

**Comment by saschaszott** [comment]
`relationship-types.xml` contains two different relationship types with identical names (`isAuthorOfPublication` and `isPublicationOfAuthor`). The names of left and right relationship types between `Publication` and `OrgUnit` should be adapted (same as in `openaire4-relationships.xml`).

**Comment by kshepherd** [comment]
@tdonohue @saschaszott @paulo-graca I have opened a PR https://github.com/DSpace/DSpace/pull/8667 which just makes the small suggested change suggested by Tim. I think the new name makes more sense, as well as guaranteeing uniqueness of those directional rel names.

**Comment by paulo-graca** [comment]
Thank you @kshepherd! But as @benbosman wrote in the PR, the model supports specifying same relation label when having different entities involved in the relationship.

I have an use case where there occurs the need to have the same relationship for different entities:
https://openaire-guidelines-for-literature-repository-managers.readthedocs.io/en/v4.0.0/field_creator.html

At the relation: isAuthorOfPublication an Author can be a Person or an Organization. 
Having a new relation like isAuthorOrgOfPublication we are saying that relation isn't the same thing, it isn't a proper Author, it is some other kind of relationship (specific for Organizations). I agree to the change of the relation if intends to describe a different purpose, like isContributorOfPublication, or isReviewerOfPublication, but not based on the entity types.

**Comment by kshepherd** [comment]
@paulo-graca I see, this does make sense if I think of it as a predicate in a linked data sense, though I admit that isn't how I'd strictly interpreted the leftward/rightward types here.

**Comment by alexandrevryghem** [comment]
I created a fix for this issue, but it does require changes to the [relationship endpoint](https://github.com/DSpace/RestContract/blob/b3d5eec5ab7cb1c4a76e2a53451a7ba8df2e0989/relationships.md#relationships-per-relationship-type), I had to add an additional parameter `entityType` to that endpoint. So I don't know if it would still be possible to fix this on 8.0, because ideally we should also have some tests for this.
